subscription-service/src/main/kotlin/com/egm/stellio/subscription/web/SubscriptionHandler.kt

```diff
@@ -64,20 +71,26 @@ class SubscriptionHandler(
     ): ResponseEntity<*> {
         val body = requestBody.awaitFirst()
         val contexts = checkAndGetContext(httpHeaders, body)
-        val parsedSubscription = parseSubscription(body, contexts)
-        checkSubscriptionNotExists(parsedSubscription).awaitFirst()
-
         val sub = getSubFromSecurityContext()
-        subscriptionService.create(parsedSubscription, sub).awaitFirst()
-        subscriptionEventService.publishSubscriptionCreateEvent(
-            parsedSubscription,
-            removeContextFromInput(body),
-            contexts
-        )
 
-        return ResponseEntity.status(HttpStatus.CREATED)
-            .location(URI("/ngsi-ld/v1/subscriptions/${parsedSubscription.id}"))
-            .build<String>()
+        return either<APiException, ResponseEntity<*>>{
+            val parsedSubscription = parseSubscription(body, contexts)
+            checkSubscriptionNotExists(parsedSubscription).awaitFirst().bind()
+
+            subscriptionService.create(parsedSubscription, sub)
```

Comment:
`subscription.create(...)` returns a `Mono`. If you don't subscribe to it in some way, it will never be executed.